### PR TITLE
Alex/refreshing

### PR DESCRIPTION
**Pull request by alexweininger**
I extracted the logic for making a request to remotion and rendering the banner into a new function. This way we can share that code between the `refreshBanner` and `bannerStreamUp` functions.

I do a "complex" (made simple via Prisma) query that fetches users who are live, are professional or personal (depending on which one we want), and have a banner enabled.

refreshBanner renders the banner, and updates their twitter. Doesn't do anything else.

For how we are scheduling the jobs, I'm using this extremely simple package that just calls a function every X period of time. For now, I'm using 10 minutes for professional. And I have the personal refreshing task/job commented out for when we verify this works.

### CODE REVIEW

File: apps/next/features/banner/bannerRefresh.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import { flipFeatureEnabled, getTwitterInfo } from "@app/util/database/postgresHelpers";
+import { getAccountsById } from "@app/util/getAccountsById";
+import { logger } from "@app/util/logger";
+import { TwitterResponseCode, updateBanner, validateTwitterAuthentication } from "@app/util/twitter/twitterHelpers";
+import { renderBanner } from "./renderBanner";
+
+export const bannerRefresh: Feature<string> = async (userId: string): Promise<string> => {
+    /**
+     * 1. Render banner image
+     * 2. Check if user is still live according to our database
```
Comment (AndrewJLang) [inline apps/next/features/banner/bannerRefresh.ts:12]: We may still have some users from the first test run that are not actually live but are in the live db. We also may hit something where they are accidently kept in the live user table. We should handle this properly, if we are talking with twitch and don't get anything back just void the change.

File: apps/next/features/banner/renderBanner.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    const bannerEntry = await getBannerEntry(userId);
+    if (bannerEntry === null) {
+        throw new Error('Could not find banner entry for user');
+    }
+
+    const authedTwitchAxios = await TwitchClientAuthService.authAxios(twitchAxios);
+
+    // get twitch stream info for user
+    // https://dev.twitch.tv/docs/api/reference#get-streams
+    const streamResponse = await authedTwitchAxios.get(`/helix/streams?user_id=${twitchUserId}`);
```
Comment (AndrewJLang) [inline apps/next/features/banner/renderBanner.ts:30]: if this is empty, they are not live. We should just void the update in this case.

File: apps/next/features/banner/renderBanner.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    foregroundProps: Record<string, unknown>;
+    backgroundProps: Record<string, unknown>;
+};
+
+/**
+ * @param userId
+ * @param twitchUserId
+ * @returns base64 of rendered banner
+ */
+export const renderBanner = async (userId: string, twitchUserId: string): Promise<string> => {
```
Comment (AndrewJLang) [inline apps/next/features/banner/renderBanner.ts:19]: This should probably also either return a specific string if they are not live or return a different type based on my comment down below.

File: apps/next/services/BannerRefreshService.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+     * @returns Queries the database and finds users who satisfy the following conditions:
+     *  - Currently live
+     *  - Active subscription of the specified product
+     *  - They have the banner feature enabled
+     */
+    private async getLiveCustomers(product: ProductName) {
+        return await this.prisma.liveStreams.findMany({
+            where: {
+                user: {
+                    // subscription: {
```
Comment (AndrewJLang) [inline apps/next/services/BannerRefreshService.ts:56]: This should be uncommented and also have another check for just professional users to start.

Comment (AndrewJLang) [reply apps/next/services/BannerRefreshService.ts:56]: Should just be uncommented I see the parameter now.

File: apps/next/util/ssr/onLoad.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+                const refreshService = new BannerRefreshService(prisma, logger);
+                await refreshService.refreshBanners('Professional');
+            },
+            (err: Error) => {
+                /* handle error here */
+                logger.error('Error occured executing scheduled job.', { ...err });
+            }
+        );
+
+        // const refreshPersonal = new AsyncTask('refresh personal',
```
Comment (AndrewJLang) [inline apps/next/util/ssr/onLoad.ts:30]: Assume we don't need this?

Comment (alexweininger) [reply apps/next/util/ssr/onLoad.ts:30]: Well I was going to uncomment it when after testing it, but maybe we should uncomment it now.

File: apps/next/features/banner/bannerRefresh.ts (hunk 1 above)
Comment (alexweininger) [reply apps/next/features/banner/bannerRefresh.ts:12]: Ok I made it return if they aren't live from the twitch api

**Pull request merged**